src/lib/protocols/quic.c

```diff
@@ -1072,7 +1099,8 @@ static const uint8_t *get_reassembled_crypto_data(struct ndpi_detection_module_s
   rc = __reassemble(flow, frag, frag_len, frag_offset,
                     &crypto_data, crypto_data_len);
   if(rc == 0) {
-    if(is_ch_complete(crypto_data, *crypto_data_len, flow->l4.udp.quic_reasm_buf_last_pos)) {
+    if(is_ch_complete(crypto_data, *crypto_data_len) &&
+      is_reasm_buf_complete(flow->l4.udp.quic_reasm_buf_bitmap, *crypto_data_len)) {
```

Comment:
Invalid read detected by valgrind
```
valgrind -v ./example/ndpiReader -t -i ./tests/pcap/quic_frags_ch_out_of_order_same_packet_craziness.pcapng
[...]
==56610== 419 errors in context 1 of 1:
==56610== Conditional jump or move depends on uninitialised value(s)
==56610==    at 0x1D3828: is_ch_complete (quic.c:1068)
==56610==    by 0x1D3828: is_ch_complete (quic.c:1062)
==56610==    by 0x1D3828: get_reassembled_crypto_data (quic.c:1102)
==56610==    by 0x1D3828: get_crypto_data (quic.c:1223)
==56610==    by 0x1D3828: ndpi_search_quic (quic.c:1639)
```

Reply:
Indeed. I zero-initialized the quic_reasm_buf and also added a missing free for the quic_reasm_buf_bitmap. I think it's fine now.